models/user_heatmap.go

```diff
@@ -32,17 +32,14 @@ func getUserHeatmapData(user *User, team *Team, doer *User) ([]*UserHeatmapData,
 		return hdata, nil
 	}
 
-	var groupBy string
-	groupByName := "timestamp" // We need this extra case because mssql doesn't allow grouping by alias
+	// Group by 15 minute intervals which will allow the client to accurately shift the timestamp to their timezone.
+	// The interval is based on the fact that there are timezones such as UTC +5:30 and UTC +12:45.
+	var groupBy = "created_unix / 900 * 900"
+	var groupByName = "timestamp" // We need this extra case because mssql doesn't allow grouping by alias
```

Comment:
I'd be careful here as the comment no longer matches the code.
MySQL requires using DIV instead of / for integer division in the group by statement,  That's fine, but what happened to not supporting grouping via an alias as the comment suggests?

FYI:
This looks like a general SQL implementation issue <https://stackoverflow.com/a/3841804>

It may be more robust to set the `groupByName` to `created_unix  / 900` or `created_unix DIV 900` and then set `groupBy` to be `groupByName + " * 900"`.  It would simplify the code.

Reply:
The comment does match the code, it matches what it used to be. M**y**SQL supports group by alias, M**S**SQL does not.

Reply:
Sorry that's my fault.  Apparently I need to move to a bigger text size.
That said, adding a comment on why mysql is special here, before that case would be good.

And I would be concerned with MSSQL possibly doing something stupid like actually dividing by 900 then multiplying by 900 when it could group after doing the divide.

Reply:
I did fix the variable declaration to be consistent with the rest of the code base.